src/components/cylc/TaskFilterSelect.vue

```diff
@@ -47,30 +48,48 @@
         v-if="index === maxVisibleStates"
         class="text-grey text-caption"
       >
-        (+{{ localValue.states.length - maxVisibleStates }})
+        (+{{ localValue.length - maxVisibleStates }})
       </span>
     </template>
   </v-select>
 </template>
 
 <script setup>
 import { computed } from 'vue'
+import { mdiClose } from '@mdi/js'
 import Task from '@/components/cylc/Task.vue'
 import Workflowicon from '@/components/cylc/gscan/WorkflowIcon.vue'
 
 const props = defineProps({
-  modelValue: { type: Object, required: true }, // { id, states }
-  items: { type: Array, required: true },
-  type: { type: String, required: true },
+  /** Array of selected states */
+  modelValue: {
+    type: Array,
+    default: () => [],
+  },
+  items: {
+    type: Array,
+    required: true,
+  },
+  type: {
+    type: String,
+    required: true,
+  },
 })
 
-defineEmits(['update:modelValue'])
+const emit = defineEmits(['update:modelValue'])
 
-const localValue = computed(() => props.modelValue)
+const localValue = computed({
+  get () {
+    return props.modelValue
+  },
+  set (val) {
+    emit('update:modelValue', val)
+  }
+})
```

Comment:
The `update:modelValue` event was not being fired so `modelValue` was never actually changing